packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(createResultSetName, 'Mutation'));
+
+      // Create a record with allowed fields, so we can test the update and delete operations.
+      const createResult1 = await user1TodoHelper.create(createResultSetName, todo, completeOwnerResultSet);
+      checkOperationResult(createResult1, { ...todo, privateContent: null, ownerContent: null }, createResultSetName);
+
+      const publicFieldSet = `
+        customId
+        publicContent
+      `;
+      // owner cannot update a record with public protected field
+      await expect(
+        async () =>
+          await user1TodoHelper.update(
+            updateResultSetName,
+            { customId: todo['customId'], publicContent: 'Public Content' },
+            publicFieldSet,
+          ),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // owner cannot update a record with a protected field that does not allow update operation
+      await expect(
+        async () => await user1TodoHelper.update(updateResultSetName, { ...todo, ownerContent: 'Owner Content' }),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // non-owner cannot update a record to re-assign ownership
+      const ownerFieldSet = `
+        customId
+        author
+      `;
+      await expect(
+        async () => await user2TodoHelper.update(updateResultSetName, { customId: todo['customId'], author: 'user2' }, ownerFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // non-owner cannot update a record with an owner protected field
+      const ownerContentFieldSet = `
+        customId
+        ownerContent
+      `;
+      await expect(
+        async () => await user2TodoHelper.update(updateResultSetName, { ...todo, ownerContent: 'Owner Content' }, ownerContentFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // owner cannot read a record with public protected field
+      const getResult1 = await user1TodoHelper.get({ customId: todo['customId'] }, publicFieldSet, false, 'all', 'customId');
+      checkOperationResult(
+        getResult1,
+        { customId: todo['customId'], publicContent: null },
+        `get${modelName}`,
+        false,
+        expectedFieldErrors(['publicContent'], modelName),
+      );
+
+      const listTodosResult1 = await user1TodoHelper.list({}, publicFieldSet, `list${modelName}`, false, 'all');
+      checkListItemExistence(listTodosResult1, `list${modelName}`, todo['customId'], true, 'customId');
+      checkListResponseErrors(listTodosResult1, expectedFieldErrors(['publicContent'], modelName, false));
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket